<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>JEPEG by dleelab</title>

    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/github-light.css">
    <script src="javascripts/scale.fix.js"></script>
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body>
    <div class="wrapper">
      <header>
        <h1 class="header">JEPEG</h1>
        <p class="header">joint effect on phenotype of eQTLs/functional SNPs associated with a gene</p>

        <ul>
          <li class="download"><a class="buttons" href="https://github.com/dleelab/jepeg/zipball/master">Download ZIP</a></li>
          <li class="download"><a class="buttons" href="https://github.com/dleelab/jepeg/tarball/master">Download TAR</a></li>
          <li><a class="buttons github" href="https://github.com/dleelab/jepeg">View On GitHub</a></li>
        </ul>

        <p class="header">This project is maintained by <a class="header name" href="https://github.com/dleelab">dleelab</a></p>


      </header>
      <section>
        <h2>
<a id="whats-new" class="anchor" href="#whats-new" aria-hidden="true"><span class="octicon octicon-link"></span></a>What's New?</h2>

<p><strong>[04/29/2015]</strong> A pre-compiled executable for JEPEG (v0.2.0) built under Linux is available now! Please see the updated output format.</p>

<p><strong>[07/22/2014]</strong> A pre-compiled executable for JEPEG (v0.1.0) built under Linux is available now!</p>

<h2>
<a id="introduction" class="anchor" href="#introduction" aria-hidden="true"><span class="octicon octicon-link"></span></a>Introduction</h2>

<p>"JEPEG is a novel software which uses only summary statistics and correlation structures estimated from a reference popluation, to i) impute the summary statistics at unmeasured eQTLs and ii) test for the joint effect of measured and imputed all eQTLs affecting a gene function/expression on a phenotype. JEPEG is useful when i) the subject-level genotype is not available and ii) fast and accurate access to gene-based p-values, pooling all association signals of unmeasured as well as measured eQTLs/functional SNPs, is required. Notably, because JEPEG uses only correlation structure estimated from a reference panel, it can be used as-is to analyze summary data coming from large pedigree studies."</p>

<h2>
<a id="citing-jepeg" class="anchor" href="#citing-jepeg" aria-hidden="true"><span class="octicon octicon-link"></span></a>Citing JEPEG</h2>

<p><strong><em>JEPEG: a summary statistics based tool for gene-level joint testing of functional variants</em></strong>, Donghyung Lee; Vernell S. Williamson; Tim B. Bigdeli; Brien P. Riley; Ayman H. Fanous; Vladimir I. Vladimirov; Silviu-Alin Bacanu; <strong><em>Bioinformatics</em></strong> (2014), <a href="http://bioinformatics.oxfordjournals.org/content/31/8/1176.long">doi:10.1093/bioinformatics/btu816</a>.</p>

<h2>
<a id="acknowledgement" class="anchor" href="#acknowledgement" aria-hidden="true"><span class="octicon octicon-link"></span></a>Acknowledgement</h2>

<p><strong>This work is supported by the National Institutes of Health with grants R25DA26119, R21MH100560 and R21AA022717.</strong></p>

<p><img src="http://upload.wikimedia.org/wikipedia/commons/thumb/6/6a/US-NIH-NIDA-Logo.svg/440px-US-NIH-NIDA-Logo.svg.png" height="100px">
<img src="http://upload.wikimedia.org/wikipedia/commons/thumb/4/4d/US-NIH-NIMH-Logo.svg/280px-US-NIH-NIMH-Logo.svg.png" height="100px">
<img src="http://upload.wikimedia.org/wikipedia/commons/thumb/7/72/US-NIH-NIAAA-Logo.svg/440px-US-NIH-NIAAA-Logo.svg.png" height="100px"></p>

<h2>
<a id="download-jepeg" class="anchor" href="#download-jepeg" aria-hidden="true"><span class="octicon octicon-link"></span></a>Download JEPEG</h2>

<p>The current release (Version 0.2.0) of JEPEG is for a Linux user. The pre-compiled executables for other operating systems (e.g.,  Windows, MacOS) will be available soon. The latest source codes of JEPEG are available upon request (dustorm_at_gmail_dot_com)</p>

<table>
<thead>
<tr>
<th align="left"><strong>Direct download link</strong></th>
<th align="left"><strong>Version</strong></th>
<th align="left"><strong>Release Date</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td align="left"><a href="https://drive.google.com/file/d/0B8CSKKDoI0UmV1Y3YVpqWlloaWs/view?usp=sharing">JEPEG for a Linux user</a></td>
<td align="left">v0.2.0</td>
<td align="left">04/29/2015</td>
</tr>
</tbody>
</table>

<h2>
<a id="download-reference-panels" class="anchor" href="#download-reference-panels" aria-hidden="true"><span class="octicon octicon-link"></span></a>Download Reference Panels</h2>

<table>
<thead>
<tr>
<th align="left"><strong>Direct download link</strong></th>
<th align="left"><strong>Number of Samples</strong></th>
<th align="left"><strong>NCBI build</strong></th>
<th align="left"><strong>Release Date</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td align="left"><a href="https://drive.google.com/file/d/0B8CSKKDoI0UmZDczT0w5OFVKSHc/edit?usp=sharing">1000 Genomes AFR Phase1 Release3</a></td>
<td align="left">246</td>
<td align="left">build 37 (hg19)</td>
<td align="left">Nov. 23 2010</td>
</tr>
<tr>
<td align="left"><a href="https://drive.google.com/file/d/0B8CSKKDoI0UmdVNlcUZFbVdFNVU/edit?usp=sharing">1000 Genomes AMR Phase1 Release3</a></td>
<td align="left">181</td>
<td align="left">build 37 (hg19)</td>
<td align="left">Nov. 23 2010</td>
</tr>
<tr>
<td align="left"><a href="https://drive.google.com/file/d/0B8CSKKDoI0UmRjdhVks2YjNVaGs/edit?usp=sharing">1000 Genomes ASN Phase1 Release3</a></td>
<td align="left">286</td>
<td align="left">build 37 (hg19)</td>
<td align="left">Nov. 23 2010</td>
</tr>
<tr>
<td align="left"><a href="https://drive.google.com/file/d/0B8CSKKDoI0UmQWF5MkRySDYyQjA/edit?usp=sharing">1000 Genomes EUR Phase1 Release3</a></td>
<td align="left">379</td>
<td align="left">build 37 (hg19)</td>
<td align="left">Nov. 23 2010</td>
</tr>
<tr>
<td align="left"><a href="">UK10K (ALSPAC and TWINSUK)</a></td>
<td align="left">2432</td>
<td align="left">build 37 (hg19)</td>
<td align="left"></td>
</tr>
</tbody>
</table>

<h2>
<a id="download-snp-annotation-data" class="anchor" href="#download-snp-annotation-data" aria-hidden="true"><span class="octicon octicon-link"></span></a>Download SNP Annotation Data</h2>

<table>
<thead>
<tr>
<th align="left"><strong>Direct download link</strong></th>
<th align="left"><strong>Version</strong></th>
<th align="left"><strong>Release Date</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td align="left"><a href="https://drive.google.com/file/d/0B8CSKKDoI0UmNVdNenRPNmtYbkk/view?usp=sharing">SNP annotations</a></td>
<td align="left">v0.2.0</td>
<td align="left">11/20/2014</td>
</tr>
<tr>
<td align="left"><a href="https://drive.google.com/file/d/0B8CSKKDoI0UmYjM1eUxMVWFiRGc/edit?usp=sharing%20SNP%20annotations">SNP annotations</a></td>
<td align="left">v0.1.0</td>
<td align="left">07/22/2014</td>
</tr>
</tbody>
</table>

<h2>
<a id="jepeg-input-file-format" class="anchor" href="#jepeg-input-file-format" aria-hidden="true"><span class="octicon octicon-link"></span></a>JEPEG input file format</h2>

<p>JEPEG takes as input a plain text file with rows and columns denoting SNPs and variables, respectively. The first line of the input file should be column names/headers. Data entries on each line should be separated by white space. </p>

<p>If your summary data set was generated based on non-imputed genotypic data and you want to impute the summary statistics of all unmeasured eQTLs based on an external reference panel (e.g. 1000 Genomes or UK10K) using DIST before JEPEG analysis, your input data must contains six columns (DIST input format): 1) rsid, 2) chromosome number, 3) base pair position, 4) reference allele, 5) alternative allele and 6) normally distributed GWAS/meta-analysis summary statistic (Z-score). JEPEG does not require the input data to be sorted in ascending order by chromosome number and base pair position or rsid. Here is a sample JEPEG input file, with 5 SNPs:</p>

<pre><code>rsid        chr  bp         a1  a2   z

rs1000109   9    117908733  C   T   -0.714464

rs10001109  4    44904653   C   T    0.721919

rs1000112   22   28273339   C   G   -0.583666

rs10001127  4    130547376  T   C    0.069329

rs1000113   5    150240076  T   C   -1.447288
</code></pre>

<p>If your input data is generated on the basis of already imputed genotype data by using genotype imputation methods such as IMPUTE2, MACH, BEAGLE, etc., you may not want to impute your data again by using DIST. You can run JEPEG without DIST imputation. In this case, you must add one more column for imputation information (info) in your input data. Here is a sample input file for this case, with 5 SNPs:</p>

<pre><code>rsid        chr  bp         a1  a2   z         info 

rs1000109   9    117908733  C   T   -0.714464  0.998

rs10001109  4    44904653   C   T    0.721919  0.731

rs1000112   22   28273339   C   G   -0.583666  0.445

rs10001127  4    130547376  T   C    0.069329  0.899

rs1000113   5    150240076  T   C   -1.447288  1
</code></pre>

<p><strong>Note:</strong> If the input data comes from a different genome assembly (e.g. NCBI build 36 (hg18)) than NCBI build 37 (hg19), it must be converted to NCBI build 37 (hg19) using a software, called <a href="https://genome.ucsc.edu/cgi-bin/hgLiftOver">liftover</a>, from UCSC.   </p>

<h2>
<a id="jepeg-output-file-format" class="anchor" href="#jepeg-output-file-format" aria-hidden="true"><span class="octicon octicon-link"></span></a>JEPEG output file format</h2>

<p>The JEPEG output file has nine columns: 1)  gene name (geneid), 2) chi-square test statistic value (chisq), 3) degrees of freedom (df), 4) JEPEG p-value (jepeg_pval), 5) number of functional SNPs associated with gene (num_snp), 6) top functional category (top_categ), 7) top category p-value (top_categ_pval), 8) top functional SNP ID (top_snp) and 9) top SNP p-value (top_snp_pval). The first line of the output file is column names/headers.  Here is a sample JEPEG output file, with 5 genes:</p>

<pre><code>geneid    chisq    df  jepeg_pval   num_snp   top_categ   top_categ_pval   top_snp      top_snp_pval

FCGR3B    14.926   1   1.80E-05     2         PFS         1.10E-04         rs5030738    5.08E-05

CSPG4P2Y  14.774   2   3.29E-05     5         TRN         1.21E-04         rs10828664   5.55E-05

SLC7A7    19.902   3   3.40E-05     33        TRN         3.51E-04         rs9304810    1.07E-03

EDARADD   13.957   2   3.93E-05     2         PFS         1.86E-04         rs966365     7.80E-06

FAM71D    13.026   1   4.34E-05     26        TAR         3.07E-04         rs45515505   1.53E-04
</code></pre>

<h2>
<a id="options" class="anchor" href="#options" aria-hidden="true"><span class="octicon octicon-link"></span></a>Options</h2>

<table>
<thead>
<tr>
<th align="left"><strong>Option</strong></th>
<th align="left"><strong>Short Flag</strong></th>
<th align="left"><strong>Parameter</strong></th>
<th align="left"><strong>Default</strong></th>
<th align="left"><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">--version</td>
<td align="left">-v</td>
<td align="left">none</td>
<td align="left">none</td>
<td align="left">Prints version information.</td>
</tr>
<tr>
<td align="left">--help</td>
<td align="left">-h</td>
<td align="left">none</td>
<td align="left">none</td>
<td align="left">Outputs a full description of all JEPEG options.</td>
</tr>
<tr>
<td align="left">--impute</td>
<td align="left">none</td>
<td align="left">none</td>
<td align="left">none</td>
<td align="left">Imputes summary statistics of unmeasured functional SNPs using DIST before JEPEG analysis.</td>
</tr>
<tr>
<td align="left">--reference</td>
<td align="left">-r</td>
<td align="left">filename</td>
<td align="left">none</td>
<td align="left">The filename of the reference population data.</td>
</tr>
<tr>
<td align="left">--referenceIndex</td>
<td align="left">-i</td>
<td align="left">filename</td>
<td align="left">none</td>
<td align="left">The filename of the reference population index data.</td>
</tr>
<tr>
<td align="left">--annotation</td>
<td align="left">-a</td>
<td align="left">filename</td>
<td align="left">none</td>
<td align="left">The filename of the SNP annotation data set.</td>
</tr>
<tr>
<td align="left">--output</td>
<td align="left">-o</td>
<td align="left">filename</td>
<td align="left">out.jepeg</td>
<td align="left">The filename of JEPEG output</td>
</tr>
<tr>
<td align="left">--windowSize</td>
<td align="left">-n</td>
<td align="left">decimal</td>
<td align="left">1.0</td>
<td align="left">The size of the DIST prediction window (Mb).</td>
</tr>
<tr>
<td align="left">--wingSize</td>
<td align="left">-m</td>
<td align="left">decimal</td>
<td align="left">0.5</td>
<td align="left">The size of the wing padded on the left and right of the DIST prediction window (Mb).</td>
</tr>
<tr>
<td align="left">--impInfoCutoff</td>
<td align="left">none</td>
<td align="left">decimal</td>
<td align="left">0.3</td>
<td align="left">The imputation information cutoff.</td>
</tr>
</tbody>
</table>

<h2>
<a id="getting-started-with-examples" class="anchor" href="#getting-started-with-examples" aria-hidden="true"><span class="octicon octicon-link"></span></a>Getting started with examples</h2>

<p>The users are required to download a pre-compiled JEPEG executable (jepeg) and a sample input file (sample.pgc1.scz.txt) from the <strong>Download JEPEG</strong> section above. The SNP annotation data set should be downloaded from <strong>Download SNP annotation data</strong> section above. A tgz-compressed directory with 1000 Genomes or UK10K reference panel datasets also should be downloaded and can be uncompressed using <strong>tar</strong> with the <strong>-zxvf</strong> options. Let’s assume that the reference panel data file and its index file named as "1kg.eur.geno.gz" and "1kg.eur.index.gz" respectively are stored in a directory /path/to/reference/ and the annotation dataset named as "jepeg.snp.annotations.v0.2.0.txt" are stored in a directory /path/to/annotation/. </p>

<p><strong>Scenario 1:</strong> Let's assume that we want to conduct JEPEG analysis after imputing unmeasured functional SNPs in our input data using DIST and then store all JEPEG results in a text file named as "jepeg.pgc1.scz.output.txt". The following command can be used for this scenario.</p>

<div class="highlight highlight-shell"><pre><span class="pl-k">&gt;&gt;</span> ./jepeg --impute sample.pgc1.scz.txt -o jepeg.pgc1.scz.output.txt -i /path/to/reference/1kg.eur.index.gz -r /path/to/reference/1kg.eur.geno.gz -a /path/to/annotation/jepeg.snp.annotations.v0.2.0.txt</pre></div>

<p>Here, we didn't set the size of the prediction window and wing by using options "--windowSize" and "--wingSize", so JEPEG will use the default size (1 Mb for the DIST prediction window and 0.5 Mb for the wing).</p>

<p><strong>Scenario 2:</strong> Here, we want to set <strong>the size of the DIST prediction window (--windowSize or -n)</strong> to be 0.6 Mb and <strong>the size of the wing padded each side of the prediction window (--wingSize or -m)</strong> to be 0.2 Mb. For this scenario, we can use the following command.</p>

<div class="highlight highlight-shell"><pre><span class="pl-k">&gt;&gt;</span> ./jepeg -o jepeg.pgc1.scz.output.txt -i /path/to/reference/1kg.eur.index.gz sample.pgc1.scz.txt -r /path/to/reference/1kg.eur.geno.gz --impute -a /path/to/annotation/jepeg.snp.annotations.v0.2.0.txt -n 0.6 -m 0.2</pre></div>

<p><strong>Scenario 3:</strong> Let's assume that our input data (summary statistics) are generated from genotype data already imputed by IMPUTE2 using 1000 Genomes European panel so we don't want to impute our input using DIST. By removing "--impute" option from the command, we can skip DIST imputation.</p>

<div class="highlight highlight-shell"><pre><span class="pl-k">&gt;&gt;</span> ./jepeg sample.input.txt -o sample.jepeg.output.txt -i /path/to/reference/1kg.eur.index.gz  -r /path/to/reference/1kg.eur.geno.gz -a /path/to/annotation/jepeg.snp.annotations.v0.2.0.txt </pre></div>
      </section>
      <footer>
        <p><small>Hosted on <a href="https://pages.github.com">GitHub Pages</a> using the Dinky theme</small></p>
      </footer>
    </div>
    <!--[if !IE]><script>fixScale(document);</script><![endif]-->
		          <script type="text/javascript">
            var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
            document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
          </script>
          <script type="text/javascript">
            try {
              var pageTracker = _gat._getTracker("UA-43996102-2");
            pageTracker._trackPageview();
            } catch(err) {}
          </script>

  </body>
</html>
